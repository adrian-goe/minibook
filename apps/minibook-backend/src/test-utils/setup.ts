import { INestApplication } from '@nestjs/common';
import request from 'supertest';
import { Test, TestingModule } from '@nestjs/testing';
import { AppModule } from '../app/app.module';

export async function gqlTestQuery(
  app: INestApplication,
  query: string,
  variables: unknown = {}
) {
  return request(app.getHttpServer()).post('/graphql').send({
    query,
    variables,
  });
}

export async function setupTestApp() {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();
  return moduleFixture.createNestApplication();
}
