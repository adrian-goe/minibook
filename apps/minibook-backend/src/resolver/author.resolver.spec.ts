import { INestApplication } from '@nestjs/common';
import { gqlTestQuery, setupTestApp } from '../test-utils/setup';

describe('Author Resolver', function () {
  let app: INestApplication;
  // let db;

  beforeAll(async () => {
    app = await setupTestApp();
    // db = getDbClient();
    await app.init();
  });

  afterAll(async () => {
    await app.close();
  });

  it('should return an author', async () => {
    const response = await gqlTestQuery(
      app,
      `
query AuthorQuery {
  author {
    id
    name
    books {
      isbn
      name
    }
  }
}
`
    );

    expect(response.status).toEqual(200);
    expect(response.body.data).toMatchSnapshot();
  });

  it('should create an author', async () => {
    const response = await gqlTestQuery(
      app,
      `
mutation Mutation($createAuthor: AuthorInput!) {
  createAuthor(createAuthor: $createAuthor) {
    id
    name
    books {
      name
      isbn
    }
  }
}
`,
      { createAuthor: { name: 'Adrian GÃ¶risch' } }
    );

    expect(response.status).toEqual(200);
    expect(response.body.data).toMatchSnapshot();
  });
});
