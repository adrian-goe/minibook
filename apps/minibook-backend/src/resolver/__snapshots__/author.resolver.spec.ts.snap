// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Resolver should return an author 1`] = `
{
  "author": {
    "books": [],
    "id": "ffed4244-9d34-4b64-9fc6-68ef6b7cc98a",
    "name": "Adrian GÃ¶risch",
  },
}
`;
